### PR TITLE
Add an extension method to Gtk.Window to have it hijack right-click

### PR DESCRIPTION
**Pull request by bratsche**
events on all Gtk.Entry widgets in its child tree and to manage
the context menu using MonoDevelop's ContextMenu component so we
have native context menus on OSX.

https://bugzilla.xamarin.com/show_bug.cgi?id=29751

### CODE REVIEW

**Comment by bratsche** [comment]
I only enabled this in one window so far, if it looks good then I'll use it in others as well.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/GtkUtil.cs [hunk 1]
```diff
@@ -470,6 +471,155 @@ static Gdk.EventKey CreateKeyEvent (uint keyval, int keyCode, Gdk.ModifierType s
 			IntPtr ptr = GLib.Marshaller.StructureToPtrAlloc (nativeEvent); 
 			return new EventKeyWrapper (ptr);
 		}
+
+		public static IEnumerable<Gtk.Widget> FindAllChildWidgets (this Gtk.Container container)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/GtkUtil.cs]: This method doesn't need to be public.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/GtkUtil.cs [hunk 2]
```diff
@@ -470,6 +471,155 @@ static Gdk.EventKey CreateKeyEvent (uint keyval, int keyCode, Gdk.ModifierType s
...
+		{
+			var widgets = new Stack<Widget> (new[] { container });
+
+			while (widgets.Any ()) {
+				var widget = widgets.Pop ();
+				yield return widget;
+
+				if (widget is Gtk.Bin) {
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/GtkUtil.cs]: Why special case? a Bin is a Container

**Comment by slluis** [comment]
LGTM

**Pull request merged**